samples/zigbee/ncp/src/main.c

```diff
@@ -9,13 +9,60 @@
  */
 
 #include <drivers/uart.h>
+#include <drivers/gpio.h>
 #include <usb/usb_device.h>
 #include <logging/log.h>
 #include <zb_nrf_platform.h>
 #include <zb_osif_ext.h>
+#include <dfu/mcuboot.h>
+#include <zephyr.h>
+#include <device.h>
 
 LOG_MODULE_REGISTER(app);
 
+#if DT_NODE_EXISTS(DT_ALIAS(rst0))
+#define RST_PIN_PORT DT_GPIO_LABEL(DT_ALIAS(rst0), gpios)
+#define RST_PIN_NUMBER DT_GPIO_PIN(DT_ALIAS(rst0), gpios)
+#define RST_PIN_LVL	0
+#endif
+
+zb_ret_t zb_osif_bootloader_run_after_reboot(void)
+{
+#if DT_NODE_EXISTS(DT_ALIAS(rst0))
+	int err = 0;
+	const struct device *rst_dev = device_get_binding(RST_PIN_PORT);
+
+	if (!rst_dev) {
+		return RET_ERROR;
+	}
+
+	err = gpio_pin_configure(rst_dev, RST_PIN_NUMBER, GPIO_OUTPUT);
+	if (err) {
+		return RET_ERROR;
+	}
+
+	/* Perform pin reset */
+	err = gpio_pin_set_raw(rst_dev, RST_PIN_NUMBER, RST_PIN_LVL);
+	if (err) {
+		return RET_ERROR;
+	}
+#endif
+	return RET_OK;
+}
+
+void zb_osif_bootloader_report_successful_loading(void)
+{
+	if (!boot_is_img_confirmed()) {
```

Comment:
Please add check for bootloader library presence e.g. `CONFIG_MCUBOOT_IMG_MANAGER`